### PR TITLE
Feature/security events rename

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Rename violations into security events

# Impacts
Everything related to violations

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [n/a] Document the feature
- [n/a] Add unit tests
- [n/a] Add acceptance tests (TestLink)

# NEWS file entries

We'll do a news entry for the whole rework, not just the renaming

### CODE REVIEW

**Comment by julsemaan** [comment]
I haven't created all the upgrade scripts yet, I'll do them in a separate PR to expedite the merge of this in order to prevent the unavoidable merge conflicts that will happen if this sits here for too long.

Please review ASAP, we should aim to get this merged by the end of the week at the latest.

**Comment by julsemaan** [comment]
BUMP, @jrouzierinverse, @cgx and @satkunas 

Please check this out, as described, this should be merged ASAP to prevent rebase issues and people working on violations outside of this

File: html/pfappserver/lib/pfappserver/Form/Config/Provisioning/opswat.pm
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Form/Config/Provisioning/opswat.pm]: security event

File: html/pfappserver/lib/pfappserver/Model/Node/Tab/SecurityEvents.pm
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Model/Node/Tab/SecurityEvents.pm]: security events

File: html/pfappserver/root/configurator/services.tt
Comment (jrouzierinverse) [inline html/pfappserver/root/configurator/services.tt]: security events

File: html/pfappserver/root/security_event/list.tt
Comment (jrouzierinverse) [inline html/pfappserver/root/security_event/list.tt]: security events

File: html/pfappserver/root/security_event/list.tt
Comment (jrouzierinverse) [inline html/pfappserver/root/security_event/list.tt]: security events

File: html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationConnectionProfiles.js
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/globals/configuration/pfConfigurationConnectionProfiles.js]: security events

File: html/pfappserver/root/static.alt/src/views/Configuration/_components/ComplianceSection.vue
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/views/Configuration/_components/ComplianceSection.vue]: Security Events

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodesSearch.vue]: Apply Security Event

File: lib/pf/cmd/pf/security_event.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/security_event.pm]: security event

File: lib/pf/cmd/pf/security_event.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/security_event.pm]: security event

File: lib/pf/cmd/pf/security_event.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/security_event.pm]: security event

File: lib/pf/cmd/pf/security_event.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/security_event.pm]: security event

File: lib/pf/cmd/pf/security_event.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf/security_event.pm]: security event

File: lib/pf/cmd/pf.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf.pm]: security event

File: lib/pf/cmd/pf.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf.pm]: security event

File: lib/pf/cmd/pf.pm
Comment (jrouzierinverse) [inline lib/pf/cmd/pf.pm]: security event

File: html/pfappserver/root/user/security_events.tt
Comment (jrouzierinverse) [inline html/pfappserver/root/user/security_events.tt]: security event

File: html/pfappserver/root/static.alt/src/views/Users/index.vue
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/views/Users/index.vue]: Security Events

File: html/pfappserver/root/static.alt/src/views/Users/index.vue
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/views/Users/index.vue]: Security Events

File: html/pfappserver/root/static.alt/src/views/Nodes/index.vue
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/views/Nodes/index.vue]: Security Events

File: html/pfappserver/root/static.alt/src/views/Nodes/index.vue
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/views/Nodes/index.vue]: Security Events

File: html/pfappserver/root/static.alt/src/globals/pfReports.js
Comment (jrouzierinverse) [inline html/pfappserver/root/static.alt/src/globals/pfReports.js]: Security Events

File: html/captive-portal/templates/emails/emails-security_event-triggered.mjml
Comment (jrouzierinverse) [inline html/captive-portal/templates/emails/emails-security_event-triggered.mjml]: Security Event

File: html/captive-portal/templates/emails/emails-security_event-triggered.mjml
Comment (jrouzierinverse) [inline html/captive-portal/templates/emails/emails-security_event-triggered.mjml]: Security Event

File: html/captive-portal/templates/emails/emails-security_event-triggered.html
Comment (jrouzierinverse) [inline html/captive-portal/templates/emails/emails-security_event-triggered.html]: Security Event

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]
add conf/security_events.conf to .gitignore

For backwards compatibility for people consuming our old api.
Maybe we should add stubs for the old names of the methods.
For example

```perl
sub trigger_violation :Public {
  return trigger_security_event(@_);
}
```

To avoid them having to rework their code twice.

**Comment by julsemaan** [comment]
I believe everything @jrouzierinverse has posted here has been addressed

**Comment by jrouzierinverse** [comment]
Everything looks good on my side.
@cgx @satkunas what do you think?

File: conf/ui.conf.es_ES
Comment (cgx) [inline conf/ui.conf.es_ES]: `# Viol.` => `# Security events` (or drop ui.conf*! 🧹)

File: conf/vlan_filters.conf.example
Comment (cgx) [inline conf/vlan_filters.conf.example]: `security_event` => `security event`

File: html/captive-portal/templates/emails/emails-security_event-triggered.html
Comment (cgx) [inline html/captive-portal/templates/emails/emails-security_event-triggered.html]: Since we no longer talk about *violations*, *infringing* should be replaced by a different word, like *related* or *associated*.

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: I would not mix snake and camel case. Use `securityEventFields`.

File: html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/_components/NodeView.vue]: I would not mix snake and camel case. Use `securityEventDescription`.

File: html/pfappserver/root/static.alt/src/views/Users/index.vue
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/index.vue]: I known those are not yet implemented, but let's fix this path using snake case: `search/open_security_events`.

File: html/pfappserver/root/static.alt/src/views/Users/index.vue
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Users/index.vue]: I known those are not yet implemented, but let's fix this path using snake case: `search/closed_security_events`.

**Review by cgx — CHANGES_REQUESTED** [review]
Shall we modify the administration guide at the same time?

File: html/captive-portal/templates/emails/emails-security_event-triggered.html
Comment (jrouzierinverse) [reply html/captive-portal/templates/emails/emails-security_event-triggered.html]: Why not just "Device Information"

**Review by satkunas — APPROVED** [review]
@julsemaan I think you caught all references. This looks good to me.

**Comment by julsemaan** [comment]
@cgx 

I'll go through your requested changes, as for the doc I left it as is for now since I'll probably rewrite a lot of sections related to security events when we have the new version implemented in the code

**Comment by julsemaan** [comment]
I think I addressed all comments made by @cgx and @jrouzierinverse 

If so, we're good to merge, if not, let me know what I missed and I'll fix it

File: html/pfappserver/root/static.alt/src/views/Users/index.vue
Comment (jrouzierinverse) [reply html/pfappserver/root/static.alt/src/views/Users/index.vue]: The paths would need to be also adjusted in the routing in pf::UnifiedApi.

Comment (jrouzierinverse) [reply html/pfappserver/root/static.alt/src/views/Users/index.vue]: We could do that after merging.

**Comment by extrafu** [comment]
@jrouzierinverse Ok to merge? If so let's do it.

**Review by cgx — APPROVED** [review]

**Comment by jrouzierinverse** [comment]
Please rebase

**Comment by julsemaan** [comment]
as the French say:

image

**Review by jrouzierinverse — APPROVED** [review]

**Pull request merged**
